eventstaticdata/main.go

```diff
@@ -14,75 +15,190 @@ import (
 // Also initialises the static data database connection
 func Init(r *mux.Router) {
 
-	initConn()
+	fetchEnvVars()
 
 	r.HandleFunc("/events", getEventsHandler).Methods("GET")
 	r.HandleFunc("/events", postEventsHandler).Methods("POST")
 	r.HandleFunc("/events/{eventId}", getEventHandler).Methods("GET")
 	r.HandleFunc("/events/{eventId}/map", postEventMapHandler).Methods("POST")
-	r.HandleFunc("/events/{eventId}/map", getEventMapHandler).Methods("GET")
 }
 
 func getEventsHandler(w http.ResponseWriter, r *http.Request) {
 
-	// TODO: handle
+	decoder := json.NewDecoder(r.Body)
+
+	var allEventsReq AllEventsRequest
+
+	err := decoder.Decode(&allEventsReq)
+
+	if err != nil {
+		log.Println(err)
+		http.Error(
+			w,
+			fmt.Sprintf("Failed to unmarshall get all events request: %s", err),
+			http.StatusBadRequest)
+		return
+	}
+
+	err = validateAllEventsRequest(&allEventsReq)
+	if err != nil {
+		log.Println(err)
+		http.Error(
+			w,
+			fmt.Sprintf("Invalid request for all Events: %s", err),
+			http.StatusBadRequest)
+		return
+	}
+
+	events, err := getAllEventsByOrganiserID(allEventsReq.OrganiserID)
+	if err != nil {
+		log.Println(err)
+		http.Error(
+			w,
+			fmt.Sprintf("Failed to get all events by organiser ID: %s", err),
+			http.StatusInternalServerError)
+		return
+	}
+
+	json.NewEncoder(w).Encode(events)
 
 }
 
 func postEventsHandler(w http.ResponseWriter, r *http.Request) {
 
 	decoder := json.NewDecoder(r.Body)
 
-	var postedEvent Event
+	var event Event
 
-	err := decoder.Decode(&postedEvent)
+	err := decoder.Decode(&event)
 	if err != nil {
 		log.Println(err)
 		http.Error(
 			w,
-			fmt.Sprint("Failed to decode event: %s", err),
+			fmt.Sprintf("Failed to unmarshall Event: %s", err),
 			http.StatusBadRequest)
 		return
 	}
 
-	err = validateEvent(&postedEvent)
+	err = validateEvent(&event)
 	if err != nil {
 		log.Println(err)
 		http.Error(
 			w,
-			fmt.Sprintf("Invalid event: %s", err),
+			fmt.Sprintf("Invalid Event: %s", err),
 			http.StatusBadRequest)
 		return
 	}
 
-	createdEvent, err := addEvent(&postedEvent)
+	err = addEvent(&event)
 	if err != nil {
 		log.Println(err)
 		http.Error(
 			w,
-			fmt.Sprintf("Failed to write event to database: %s", err),
-			http.StatusBadRequest)
+			fmt.Sprintf("Failed to write Event to database: %s", err),
+			http.StatusInternalServerError)
 		return
 	}
 
-	json.NewEncoder(w).Encode(createdEvent)
+	json.NewEncoder(w).Encode(event)
 
 }
 
 func getEventHandler(w http.ResponseWriter, r *http.Request) {
 
-	// TODO: handle
+	vars := mux.Vars(r)
+	idStr := vars["eventId"]
```

Comment:
With maps you can add an extra `ok` value:
```suggestion
	idStr, ok := vars["eventId"]
```
and then write
```
   if !ok {
     ...
```
Do you think that looks cleaner?